### PR TITLE
Update vtkm pkg new release

### PR DESCRIPTION
**Pull request by vicentebolea**
Finally! After a quiet year, VTK-m v1.6.0 first RC is out  :tada: :tada: :tada:

Exciting new [features](https://gitlab.kitware.com/vtk/vtk-m/-/tags/v1.6.0-rc1) ready to be used.

In addition that, this PR also:

- Adds a release branch in the spack package.
- Sets v.1.5.1 as the preferred version which will be changed after the final 1.6.0 release is out.

Let me know if we need to change anything else in the package

ViB

### CODE REVIEW

**Comment by vicentebolea** [comment]
@kmorel

File: var/spack/repos/builtin/packages/vtk-m/package.py [hunk 1]
```diff
@@ -25,11 +25,13 @@ class VtkM(CMakePackage, CudaPackage):
...
 
     version('master', branch='master')
+    version('release', branch='release')
+    version('1.6.0-rc1', sha256="223f23b11260618c19c6a4fdc10e235f3f14fe955f604e55a068dabddd31af95")
     version('1.5.5', commit="d2d1c854adc8c0518802f153b48afd17646b6252")
     version('1.5.4', commit="bbba2a1967b271cc393abd043716d957bca97972")
     version('1.5.3', commit="a3b8525ef97d94996ae843db0dd4f675c38e8b1e")
     version('1.5.2', commit="c49390f2537c5ba8cf25bd39aa5c212d6eafcf61")
-    version('1.5.1', sha256="64c19e66c0d579cfb21bb0df10d649b523b470b0c9a6c2ea5fd979dfeda2c25e")
+    version('1.5.1', sha256="64c19e66c0d579cfb21bb0df10d649b523b470b0c9a6c2ea5fd979dfeda2c25e", default=True)
```
Comment (kmorel) [inline var/spack/repos/builtin/packages/vtk-m/package.py]: I assume the `default=True` is supposed to set the preferred or default version to install. However, when I run `spack info vtk-m`, it says the preferred version is `1.6.0-rc1`. And it looks like it by default installs that version.

Comment (vicentebolea) [reply var/spack/repos/builtin/packages/vtk-m/package.py]: Good catch, find this fixed in the last push.

**Review by alalazo — APPROVED** [review]

**Comment by cyrush** [comment]
@vicentebolea preferred change back to 1.5.1 makes it so ascent doesn't build out of the box with spack. Perhaps that's fine b/c the out-of-the-box stable period has been historically short -- but was there a use case required 1.5.1 to be preferred?

**Comment by cyrush** [comment]
ah, i guess this was b/c 1.6.0-rc1 implicitly became the default?

**Comment by vicentebolea** [comment]
> ah, i guess this was b/c 1.6.0-rc1 implicitly became the default?

@cyrush yes, we have to choose a new preferred release until 1.6.0 is out. I defaulted to 1.5.1 since it was the last official VTKm main repo. The other bugfixes releases in Spark 1.5.2-1.5.5 are specific commits in masters which I believe that one of our colleague ensured that they can build in Ascent.

I do not feel very strong about reverting to 1.5.1 since it will be temporal change until we get 1.6.0 out. Later, we are changing the release workflow in a way that we will have more emphasis on keep maintaining releases which are universally supported.

In fact I will remove that change, it was brought for consistency but I can see that it can be bothersome to some of our users.

@cyrush Thanks to point it out.

**Pull request merged**